module-system/sov-modules-api/src/gas.rs

```diff
@@ -0,0 +1,92 @@
+use anyhow::Result;
+
+use core::fmt;
+
+/// A gas unit that provides scalar conversion from complex, multi-dimensional types.
+pub trait GasUnit: fmt::Debug + Clone {
+    /// A zeroed instance of the unit.
+    const ZEROED: Self;
+
+    /// Creates a unit from a multi-dimensional unit with arbitrary dimension.
+    fn from_arbitrary_dimensions(dimensions: &[u64]) -> Self;
```

Comment:
This function will allow us to convert arbitrary arrays from the JSON file into the concrete, chosen type.

It is designed to have const-like performance, as in the link below
https://rust.godbolt.org/z/rPhaxnPEY